File: DesignPatterns/Creational/Factory/simple_factory/go/algorithm/old_tax_algorithm.go
package algorithm

import (
	"fmt"
	"simple_factory/contracts"
)

type OldTaxAlgorithm struct {
}

func (ot *OldTaxAlgorithm) CalculateTax(salary_details *contracts.SalaryDetails) float64 {
	fmt.Println("salary_details: ", salary_details)
	return 2.5*salary_details.Base + 3.3*salary_details.HRA
}

File: DesignPatterns/Creational/Factory/simple_factory/python/old_tax_algorithm.py
from salary_details import SalaryDetails
from tax_algorithm import TaxAlgorithm

class OldTaxAlgorithm(TaxAlgorithm):
    def __init__(self):
        super().__init__()

    def calculateTax(self, salary_details: SalaryDetails) -> float:
        tax = 2.4 * salary_details.base + 3.3 * salary_details.hra
        return tax